Sources/SwiftSyntaxBuilder/Token.swift

```diff
@@ -14,34 +11,65 @@
 //===----------------------------------------------------------------------===//
 
 import SwiftSyntax
-extension TokenSyntax: ExpressibleAsTokenList, ExpressibleAsNonEmptyTokenList, ExpressibleAsBinaryOperatorExpr, ExpressibleAsDeclModifier, ExpressibleAsIdentifierExpr {
+
+/// Represents `TokenSyntax` in `SwiftSyntaxBuilder`.
+/// At the moment, this just wraps `TokenSyntax`, but we can make it store just the information necessary to build a `TokenSyntax` in the future.
+public struct Token: SyntaxBuildable, ExpressibleAsTokenList, ExpressibleAsNonEmptyTokenList, ExpressibleAsBinaryOperatorExpr, ExpressibleAsDeclModifier, ExpressibleAsIdentifierExpr {
+  let tokenSyntax: TokenSyntax
+
+  var text: String {
+    return tokenSyntax.text
+  }
+
+  init(tokenSyntax: TokenSyntax) {
+    self.tokenSyntax = tokenSyntax
+  }
+
+  public func withLeadingTrivia(_ leadingTrivia: Trivia) -> Token {
+    return Token(tokenSyntax: tokenSyntax.withLeadingTrivia(leadingTrivia))
+  }
+
+  public func withTrailingTrivia(_ leadingTrivia: Trivia) -> Token {
+    return Token(tokenSyntax: tokenSyntax.withTrailingTrivia(leadingTrivia))
+  }
+
+  public func buildToken() -> TokenSyntax {
+    return tokenSyntax
+  }
+
+  public func buildSyntax(format: Format, leadingTrivia: Trivia?) -> Syntax {
+    return Syntax(tokenSyntax)
+  }
+
   /// Conformance to ExpressibleAsTokenList
   public func createTokenList() -> TokenList {
     return TokenList([self])
   }
+
   /// Conformance to ExpressibleAsNonEmptyTokenList
   public func createNonEmptyTokenList() -> NonEmptyTokenList {
     return NonEmptyTokenList([self])
   }
+
   /// Conformance to ExpressibleAsBinaryOperatorExpr
   public func createBinaryOperatorExpr() -> BinaryOperatorExpr {
     return BinaryOperatorExpr(operatorToken: self)
   }
+
   /// Conformance to ExpressibleAsDeclModifier
   public func createDeclModifier() -> DeclModifier {
     return DeclModifier(name: self)
   }
+
   /// Conformance to ExpressibleAsIdentifierExpr
   public func createIdentifierExpr() -> IdentifierExpr {
     return IdentifierExpr(identifier: self)
   }
```

Comment:
Since we no longer treat `TokenSyntax` specially, could we generate this file (along with these conformances)?

Reply:
Good idea. I generated it.